File: JavaScript/recursiveBinarySearch.js
let numbers = [1, 2, 3, 4, 5, 6, 7];

function recursiveBinarySearch(arr, target) {
  if (arr.length === 0) return false;

  let mid = Math.floor(arr.length / 2);

  if (arr[mid] === target) return true;
  else if (arr[mid] < target) {
    return recursiveBinarySearch(arr.slice(mid + 1), target);
  } else {
    return recursiveBinarySearch(arr.slice(0, mid), target);
  }
}

console.log(recursiveBinarySearch(numbers, 12));
console.log(recursiveBinarySearch(numbers, 4));

File: Python/recursive_binary_search.py
def recursive_binary_search(list,target):
    if len(list) == 0:
        return False
    else:
        midpoint = len(list)//2
        if list[midpoint] == target:
            return True
        else:
            if list[midpoint] < target:
                return recursive_binary_search(list[midpoint + 1:],target)
            else:
                return recursive_binary_search(list[:midpoint],target)
            

numbers = [1,2,3,4,5,6,7]
print(recursive_binary_search(numbers,12))
print(recursive_binary_search(numbers,4))
